### PR TITLE
Fatal error message is shown when cancelling Google login

### LINKED ISSUE #1185: fatal error message is shown when cancelling google login
1. Open app
2. Tap `Continue with Gmail`
3. Tap `Cancel` in the modal
4. 🚨BUG - Fatal error screen is shown

No error screen should be shown there, as user just cancelled login attempt.

https://user-images.githubusercontent.com/1062093/144913090-1f6805a3-49d7-4af9-924a-18ef2ea31c6a.mp4

### PR DESCRIPTION
**Pull request by ivan-ushakov**
close #1185

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests will be added later (issue #1186)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: FlowCrypt/Functionality/Services/GlobalRouter.swift [hunk 1]
```diff
@@ -95,7 +95,7 @@ extension GlobalRouter: GlobalRouterType {
             }
         } catch {
             logger.logError("Failed to sign out due to \(error.localizedDescription)")
-            handle(error: error, appContext: appContext)
+            hanleFatalError(error)
```
Comment (ivan-ushakov) [inline FlowCrypt/Functionality/Services/GlobalRouter.swift:98]: @tomholub
Fatal error or splash screen?

File: FlowCrypt/Functionality/Services/GlobalRouter.swift [hunk 2]
```diff
@@ -134,7 +134,7 @@ extension GlobalRouter: GlobalRouterType {
             proceed(with: appContext.withSession(session))
         } catch {
             logger.logError("Failed to switch active user due to \(error.localizedDescription)")
-            handle(error: error, appContext: appContext)
+            hanleFatalError(error)
```
Comment (ivan-ushakov) [inline FlowCrypt/Functionality/Services/GlobalRouter.swift:137]: Same here. What if we can not switch user because of DB error?

Comment (tomholub) [reply FlowCrypt/Functionality/Services/GlobalRouter.swift:137]: modal + stay on current VC

File: FlowCrypt/Functionality/Services/GlobalRouter.swift (hunk 1 above)
Comment (tomholub) [reply FlowCrypt/Functionality/Services/GlobalRouter.swift:98]: modal to show the message content, then remain on previous VC?

**Review by tomholub — CHANGES_REQUESTED** [review]
Marking with changes requested as above, for clarity

File: FlowCrypt/Functionality/Services/GlobalRouter.swift (hunk 2 above)
Comment (ivan-ushakov) [reply FlowCrypt/Functionality/Services/GlobalRouter.swift]: Let's throw it

File: FlowCrypt/Functionality/Services/GlobalRouter.swift (hunk 1 above)
Comment (ivan-ushakov) [reply FlowCrypt/Functionality/Services/GlobalRouter.swift]: Let's throw it

**Comment by ivan-ushakov** [comment]
`SendEmailToRecipientWithoutPublicKey.spec.ts` works locally, so I run UI tests again

**Review by tomholub — APPROVED** [review]

**Pull request merged**